docker/k8s_tools.py

```diff
@@ -35,45 +35,68 @@ def wait_pods_running(label_selector, desired):
 
 
 def count_pods_by_phase(label_selector, phase):
-    pod_list = fetch_pods_info(label_selector)
-    filtered_pod_list = filter(lambda x: x[0] == phase, pod_list)
-    return len(filtered_pod_list)
+    pod_list = fetch_pods_info(label_selector, phase)
+    return len(pod_list)
 
 
-def fetch_pserver_ips():
-    label_selector = "paddle-job-pserver=%s" % PADDLE_JOB_NAME
-    pod_list = fetch_pods_info(label_selector)
-    pserver_ips = [item[1] for item in pod_list]
-    return ",".join(pserver_ips)
+def fetch_endpoints(label_selector, port=None, phase=None):
+    pod_list = fetch_pods_info(label_selector, phase)
+    ips = [item[1] for item in pod_list]
+    ips.sort()
 
+    r = []
+    for ip in ips:
+        if port != None:
+            ip = "{0}:{1}".format(ip, port)
 
-def fetch_master_ip():
-    label_selector = "paddle-job-master=%s" % PADDLE_JOB_NAME
-    pod_list = fetch_pods_info(label_selector)
-    master_ips = [item[1] for item in pod_list]
-    return master_ips[0]
+        r.append(ip)
 
+    return ",".join(r)
 
-def fetch_trainer_id():
-    label_selector = "paddle-job=%s" % PADDLE_JOB_NAME
-    pod_list = fetch_pods_info(label_selector)
-    trainer_ips = [item[1] for item in pod_list]
-    trainer_ips.sort()
+
+def fetch_id(label_selector, phase=None):
+    ips = fetch_endpoints(label_selector, phase=phase)
     local_ip = socket.gethostbyname(socket.gethostname())
-    for i in xrange(len(trainer_ips)):
-        if trainer_ips[i] == local_ip:
+    for i in xrange(len(ips)):
+        if ips[i] == local_ip:
             return i
     return None
 
 
+def fetch_trainer_ips(label_selector, port=None, phase=None):
+    return fetch_endpoints(label_selector, port, phase)
+
+
+def fetch_pserver_ips(label_selector, port=None, phase=None):
```

Comment:
I'm confused `fetch_pserver_ips` called `fetch_endpoints` function, `fetch_endpoints ` would return `IP:PORT` rather than `IP`.

Reply:
`fetch_xxx_ips` can have no argument `port` because we always use `None`. And, maybe it's better to add some comments to tell if `port=None`, `fetch_endpoints` will return a list of IPs.

Reply:
Done.